src/style-property-map.js

```diff
@@ -14,16 +14,18 @@
 
 (function(shared, scope, testing) {
 
-function StylePropertyMap() {}
+function StylePropertyMap(styleObject) {
+  this._styleObject = styleObject;
+}
 
-  StylePropertyMap.prototype = {
-    append: function(property, value) {},
-    delete: function(property) {},
-    get: function(property) {},
-    getAll: function(property) {},
-    has: function(property) {},
-    set: function(property, value) {},
-    getProperties: function() {}
+  StylePropertyMap.prototype.prototype = StylePropertyMapReadOnly.prototype;
```

Comment:
This is what I told him to do. But it was actually wrong, it should have been 
```
StylePropertyMap.prototype.__proto__ = StylePropertyMapReadOnly.prototype;
```
Sorry Nick.

But yes, using the inherit method is better anyway because it fixed the constructor.